File: frontend/src/services/customerService.js
import api from './api';

export const getAllCustomers = () => api.get('/customers');
export const createCustomer = (data) => api.post('/customers', data);
export const updateCustomer = (id, data) => api.put(`/customers/${id}`, data);
export const deleteCustomer = (id) => api.delete(`/customers/${id}`);

File: src/main/java/com/pos/pos/service/CustomerService.java
package com.pos.pos.service;

import com.pos.pos.dto.CustomerDto;
import java.util.List;

public interface CustomerService {
    CustomerDto createCustomer(CustomerDto customerDto);
    CustomerDto getCustomerById(Long id);
    List<CustomerDto> getAllCustomers();
    CustomerDto updateCustomer(CustomerDto customerDto, Long id);
    void deleteCustomer(Long id);
}
